### PR TITLE
redolog: add redo log into cdc core logic, add integration tests

### LINKED ISSUE #3117: redolog: gc in capture node not working if not owner role
### What did you do?

enable redolog in config

### What did you expect to see?

redo logs should be enable gc base on some rules

### What did you see instead?

gc is not working at capture node if not in owner role

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
5.2
```

TiCDC version (execute `cdc version`):

```console
redolog dev branch
```

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

part of #2351 redo log based replication to implement eventually consistency

#2685 contains all of the change for the requirement, will split into small ones for review, this is one of the prs, split from the #2685, this one contains #2809,#2810,#2811,#2812,#2814 review #2809,#2810,#2811,#2812,#2814 first. Or review commit https://github.com/amyangfei/ticdc/commit/331d9bd270cada531a62b25be4cf12d5dca7ef32 directly

Also close #3117

### What is changed and how it works?

- Add redo log manager into cdc core logic, the workflow is as follows

  image1

- Add integration tests for redo log writer and reader

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sink/manager_test.go [hunk 1]
```diff
@@ -188,17 +188,19 @@ func (s *managerSuite) TestManagerAddRemoveTable(c *check.C) {
 		}
 	}
 
+	redoManager := redo.NewDisabledManager()
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
 		// add three table and then remote one table
-		for i := 0; i < goroutineNum; i++ {
+		for i := 0; i < 200; i++ {
```
Comment (hi-rustin) [inline cdc/sink/manager_test.go:196]: Why don't we keep using it?

File: tests/consistent_replicate_local/run.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		return
+	fi
+	echo "global resolved ts $rts not forward to $check_tso"
+	exit 1
+}
+
+export -f check_resolved_ts
+
+function run() {
+	# TODO: sink kafka sink, since this is a MySQL eventual consistent test
```
Comment (hi-rustin) [inline tests/consistent_replicate_local/run.sh:27]: Sorry, I don't quite understand the comment?

**Review by hi-rustin — APPROVED** [review]
rest LGTM

File: cdc/sink/manager_test.go (hunk 1 above)
Comment (amyangfei) [reply cdc/sink/manager_test.go:196]: revert it, maybe caused by merge conflict.

Comment (amyangfei) [inline cdc/sink/manager_test.go:196]:
```suggestion
		for i := 0; i < goroutineNum; i++ {
```

File: tests/consistent_replicate_local/run.sh (hunk 2 above)
Comment (amyangfei) [inline tests/consistent_replicate_local/run.sh:27]:
```suggestion
	# we only support eventual consistent replication with MySQL sink
```

File: tests/consistent_replicate_s3/run.sh [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		return
+	fi
+	echo "global resolved ts $rts not forward to $check_tso"
+	exit 1
+}
+
+export -f check_resolved_ts
+
+function run() {
+	# TODO: sink kafka sink, since this is a MySQL eventual consistent test
```
Comment (amyangfei) [inline tests/consistent_replicate_s3/run.sh:62]:
```suggestion
	# we only support eventually consistent replication with MySQL sink
```

File: tests/consistent_replicate_local/run.sh (hunk 2 above)
Comment (amyangfei) [reply tests/consistent_replicate_local/run.sh]: Update the comment, at first I meant in the future, we may support eventually consistent replication with the link TiCDC -> Kafka -> TiDB/MySQL.
However we only need to support eventually consistent with MySQL sink now, so I changed the comment according to the real situation

**Pull request merged**